bundles/azure-bundle/src/main/java/org/apache/gravitino/abs/credential/ADLSTokenProvider.java

```diff
@@ -0,0 +1,136 @@
+  public void initialize(Map<String, String> properties) {
+    ADLSCredentialConfig adlsCredentialConfig = new ADLSCredentialConfig(properties);
+    this.storageAccountName = adlsCredentialConfig.storageAccountName();
+    this.tenantId = adlsCredentialConfig.tenantId();
+    this.clientId = adlsCredentialConfig.clientId();
+    this.clientSecret = adlsCredentialConfig.clientSecret();
+    this.endpoint =
+        String.format("https://%s.%s", storageAccountName, ADLSTokenCredential.ADLS_DOMAIN);
+    this.tokenExpireSecs = adlsCredentialConfig.tokenExpireInSecs();
+  }
+
+  @Override
+  public void close() {}
+
+  @Override
+  public String credentialType() {
+    return CredentialConstants.ADLS_TOKEN_CREDENTIAL_PROVIDER_TYPE;
+  }
+
+  @Override
+  public Credential getCredential(CredentialContext context) {
+    if (!(context instanceof PathBasedCredentialContext)) {
+      return null;
+    }
+    PathBasedCredentialContext pathBasedCredentialContext = (PathBasedCredentialContext) context;
+
+    Set<String> writePaths = pathBasedCredentialContext.getWritePaths();
+    Set<String> readPaths = pathBasedCredentialContext.getReadPaths();
+
+    Set<String> combinedPaths = new HashSet<>(writePaths);
+    combinedPaths.addAll(readPaths);
+
+    if (combinedPaths.size() != 1) {
+      throw new IllegalArgumentException(
+          "ADLS should contain exactly one unique path, but found: "
+              + combinedPaths.size()
+              + " paths: "
+              + combinedPaths);
+    }
+    String uniquePath = combinedPaths.iterator().next();
+
+    ClientSecretCredential clientSecretCredential =
+        new ClientSecretCredentialBuilder()
+            .tenantId(tenantId)
+            .clientId(clientId)
+            .clientSecret(clientSecret)
+            .build();
+
+    DataLakeServiceClient dataLakeServiceClient =
+        new DataLakeServiceClientBuilder()
+            .endpoint(endpoint)
+            .credential(clientSecretCredential)
+            .buildClient();
+
+    OffsetDateTime start = OffsetDateTime.now();
+    OffsetDateTime expiry = start.plusSeconds(tokenExpireSecs);
+    UserDelegationKey userDelegationKey = dataLakeServiceClient.getUserDelegationKey(start, expiry);
+
+    PathSasPermission pathSasPermission =
+        new PathSasPermission()
+            .setReadPermission(true)
+            .setWritePermission(true)
+            .setDeletePermission(true)
+            .setListPermission(true)
+            .setCreatePermission(true)
+            .setAddPermission(true);
+
+    DataLakeServiceSasSignatureValues signatureValues =
+        new DataLakeServiceSasSignatureValues(expiry, pathSasPermission);
+
+    ADLSLocationUtils.ADLSLocationParts locationParts = ADLSLocationUtils.parseLocation(uniquePath);
+    String sasToken =
+        new DataLakeSasImplUtil(
+                signatureValues,
+                locationParts.getContainer(),
+                locationParts.getPath().replaceAll("^/+|/*$", ""),
```

Comment:
please provide a separate function to do `replaceAll("^/+|/*$", "")`, add comments and test to it